internal/featuretests/v3/gatewayapi_test.go

```diff
@@ -30,40 +30,60 @@ import (
 	gatewayapi_v1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
 )
 
-var gateway = &gatewayapi_v1alpha1.Gateway{
-	ObjectMeta: metav1.ObjectMeta{
-		Name:      "contour",
-		Namespace: "projectcontour",
-	},
-	Spec: gatewayapi_v1alpha1.GatewaySpec{
-		Listeners: []gatewayapi_v1alpha1.Listener{{
-			Port:     80,
-			Protocol: "HTTP",
-			Routes: gatewayapi_v1alpha1.RouteBindingSelector{
-				Namespaces: &gatewayapi_v1alpha1.RouteNamespaces{
-					From: routeSelectTypePtr(gatewayapi_v1alpha1.RouteSelectAll),
+var (
+	gc = &gatewayapi_v1alpha1.GatewayClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "contour",
+		},
+		Spec: gatewayapi_v1alpha1.GatewayClassSpec{
+			Controller: "projectcontour.io/contour",
+		},
+		Status: gatewayapi_v1alpha1.GatewayClassStatus{
```

Comment:
Should we leave status off of the object so that we can verify that functionality works as expected?

Reply:
I tried removing status and setting "Scheduled=false" (default) but the test fails. This works when I run the e2e tests and when I test against a running cluster. Can you tell if I need to change something in `setup()`?

Reply:
As long as we have a way to check that Gateways that point to not-admitted GatewayClasses don't work, then I think this is okay.